packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -0,0 +1,951 @@
+    if (onPressed != null && (isSelected[index] || (index != 0 && isSelected[index - 1]))) {
+      return BorderSide(
+        color: selectedBorderColor ?? toggleButtonsTheme.selectedBorderColor ?? theme.colorScheme.primary,
+        width: resultingBorderWidth,
+      );
+    } else if (onPressed != null && !isSelected[index]) {
+      return BorderSide(
+        color: borderColor ?? toggleButtonsTheme.borderColor ?? theme.colorScheme.onSurface,
+        width: resultingBorderWidth,
+      );
+    } else {
+      return BorderSide(
+        color: disabledBorderColor ?? toggleButtonsTheme.disabledBorderColor ?? theme.disabledColor,
+        width: resultingBorderWidth,
+      );
+    }
+  }
+
+  BorderSide _getHorizontalBorderSide(
+    int index,
+    ThemeData theme,
+    ToggleButtonsThemeData toggleButtonsTheme,
+  ) {
+    if (!renderBorder)
+      return BorderSide.none;
+
+    final double resultingBorderWidth = borderWidth ?? toggleButtonsTheme.borderWidth ?? _defaultBorderWidth;
+    if (onPressed != null && isSelected[index]) {
+      return BorderSide(
+        color: selectedBorderColor ?? toggleButtonsTheme.selectedBorderColor ?? theme.colorScheme.primary,
+        width: resultingBorderWidth,
+      );
+    } else if (onPressed != null && !isSelected[index]) {
+      return BorderSide(
+        color: borderColor ?? toggleButtonsTheme.borderColor ?? theme.colorScheme.onSurface,
+        width: resultingBorderWidth,
+      );
+    } else {
+      return BorderSide(
+        color: disabledBorderColor ?? toggleButtonsTheme.disabledBorderColor ?? theme.disabledColor,
+        width: resultingBorderWidth,
+      );
+    }
+  }
+
+  BorderSide _getTrailingBorderSide(
+    int index,
+    ThemeData theme,
+    ToggleButtonsThemeData toggleButtonsTheme,
+  ) {
+    if (!renderBorder)
+      return BorderSide.none;
+
+    if (index != children.length - 1)
+      return null;
+
+    final double resultingBorderWidth = borderWidth ?? toggleButtonsTheme.borderWidth ?? _defaultBorderWidth;
+    if (onPressed != null && (isSelected[index])) {
+      return BorderSide(
+        color: selectedBorderColor ?? toggleButtonsTheme.selectedBorderColor ?? theme.colorScheme.primary,
+        width: resultingBorderWidth,
+      );
+    } else if (onPressed != null && !isSelected[index]) {
+      return BorderSide(
+        color: borderColor ?? toggleButtonsTheme.borderColor ?? theme.colorScheme.onSurface,
+        width: resultingBorderWidth,
+      );
+    } else {
+      return BorderSide(
+        color: disabledBorderColor ?? toggleButtonsTheme.disabledBorderColor ?? theme.disabledColor,
+        width: resultingBorderWidth,
+      );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(
+      !isSelected.any((bool val) => val == null),
+      'There is a null value in isSelected: $isSelected'
+    );
+    assert(() {
```

Comment:
Same comment as before: the assert statement cannot be placed in the constructor because it is a const constructor. Doing so triggers an analyzer error